collector.go

```diff
@@ -123,6 +138,10 @@ func (cc *ChunkedCollector) Flush() error {
 		if err := cc.Collector.Collect(spanID, p...); err != nil {
```

Comment:
Since I'm not familiar with it, under what conditions will `Collect` return? What guarantees does it have?

Is it possible that it will take a really long time and almost never return, effectively ignoring any `FlushTimeout` value?